### PR TITLE
Bindings: Add `ProtocolParameters` tests

### LINKED ISSUE #1338: Bindings ProtocolParameters tests
## Description

Similar to https://github.com/iotaledger/iota-sdk/blob/2.0/sdk/tests/types/protocol.rs

As requested here: https://github.com/iotaledger/iota-sdk/pull/1276#pullrequestreview-1635667286

## Motivation

We want to validate that the node sends us the protocol parameters we expect on all bindings

## Requirements

- [x] Nodejs
- [x] Python

## Open questions

- What is the purpose of this test in bindings after the protocol parameters are finalized?

### PR DESCRIPTION
**Pull request by Alex6323**
## Links to any relevant issues

Closes #1338

### CODE REVIEW

File: bindings/nodejs/tests/types/protocol_parameters.spec.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    it('creates ProtocolParameters from a fixture', async () => {
+
+        // TODO: is there a way around `as unknown`? The string->bigint parameters are the issue.
+        const params = protocol_parameters.params as unknown as ProtocolParameters;
+
+        // Check whether parameters were correctly set
+        expect(params.type).toEqual(0);
+        expect(params.version).toEqual(3);
+        expect(params.networkName).toEqual("TestJungle");
+        expect(params.bech32Hrp).toEqual("tgl");
```
Comment (thibault-martinez) [inline bindings/nodejs/tests/types/protocol_parameters.spec.ts]: I don't think we need to compare all the fields btw, we just need to compare the hash and it'll automatically include all the fields

File: bindings/nodejs/tests/types/protocol_parameters.spec.ts [hunk 2]
```diff
@@ -4,34 +4,25 @@
...
-        expect(params.networkName).toEqual("TestJungle");
-        expect(params.bech32Hrp).toEqual("tgl");
+    it('compares ProtocolParameters bytes from a fixture', async () => {
+        // TODO: serialize to ProtocolParameters to bytes
+    });
 
-        // TODO: is there a way around `as unknown`? The string->bigint parameters are the issue.
-        const storageScoreParams = protocol_parameters.params.storageScoreParameters as unknown as StorageScoreParameters;
+    it('compares ProtocolParameters hash from a fixture', async () => {
+        // TODO: do we want to hash from binding side? Feels unnecessary tbh.
```
Comment (thibault-martinez) [inline bindings/nodejs/tests/types/protocol_parameters.spec.ts]: no we don't

File: bindings/nodejs/tests/types/protocol_parameters.spec.ts [hunk 3]
```diff
@@ -4,34 +4,25 @@
...
-        // TODO: is there a way around `as unknown`? The string->bigint parameters are the issue.
-        const params = protocol_parameters.params as unknown as ProtocolParameters;
-
-        // Check whether parameters were correctly set
-        expect(params.type).toEqual(0);
-        expect(params.version).toEqual(3);
-        expect(params.networkName).toEqual("TestJungle");
-        expect(params.bech32Hrp).toEqual("tgl");
+    it('compares ProtocolParameters bytes from a fixture', async () => {
+        // TODO: serialize to ProtocolParameters to bytes
```
Comment (thibault-martinez) [inline bindings/nodejs/tests/types/protocol_parameters.spec.ts]: Not necessary

File: bindings/nodejs/lib/types/utils/bridge/utils.ts [hunk 4]
```diff
@@ -145,6 +145,13 @@ export interface __IsAddressValidMethod__ {
     };
 }
 
+export interface __ProtocolParametersHashMethod__ {
+    name: 'protocolParametersHash';
+    data: {
+        protocol_parameters: ProtocolParameters;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/utils/bridge/utils.ts]: camelCase

File: bindings/core/src/method/utils.rs [hunk 5]
```diff
@@ -127,6 +127,12 @@ pub enum UtilsMethod {
         serial_number: u32,
         token_scheme_type: u8,
     },
+    /// Computes the hash of the given protocol parameters.
+    #[serde(rename_all = "camelCase")]
+    ProtocolParametersHash {
+        /// Network Protocol Parameters
```
Comment (thibault-martinez) [inline bindings/core/src/method/utils.rs]: Tbh I wouldn't add such comments, I know there are already some but I'll probably clean it us and remove at some point. This is not a public crate, we don't care.

File: bindings/nodejs/tests/types/protocol_parameters.spec.ts (hunk 1 above)
Comment (Alex6323) [reply bindings/nodejs/tests/types/protocol_parameters.spec.ts]: Fixed

**Review by Thoralf-M — DISMISSED** [review]

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
Fmt

**Review by thibault-martinez — APPROVED** [review]

File: sdk/tests/types/fixtures/protocol_parameters.json
Comment (DaughterOfMars) [inline sdk/tests/types/fixtures/protocol_parameters.json:1]: Can we put comments in this file? And if so, can we put the link to the source of the data here?

**Review by Thoralf-M — APPROVED** [review]

File: sdk/tests/types/fixtures/protocol_parameters.json
Comment (thibault-martinez) [reply sdk/tests/types/fixtures/protocol_parameters.json:1]: I don't think we can comment in JSON files ?

Comment (Alex6323) [reply sdk/tests/types/fixtures/protocol_parameters.json:1]: seems like it's only possible with JSON5

Comment (DaughterOfMars) [reply sdk/tests/types/fixtures/protocol_parameters.json:1]: looks like serde_json doesn't accept comments, bummer

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**